### ISSUE TITLE
Issue in iOS9

### ISSUE THREAD

**Issue by subinrevi** [BranchMetrics/iOS-Deferred-Deep-Linking-SDK#256]
Hi,
     Clicking on the link doesn't take the user to the app installed in the phone in iOS9. I tested it using the test_key but when i changed the key to live its not working in iOS9. I have also changed the bundle id (have made the associated bundle id changes in dashboard and info.plist as well)

**Comment by aaustin** [comment]
Hey @subinrevi - There is one mistake that most people are making. If you have a custom subdomain configured in the Branch dash, you need to enable TLS on it. Currently, we've only figured out how to do this with Cloudflare.
https://dev.branch.io/recipes/branch_universal_links/ios/#advanced-support-ssltls-with-your-domain-for-universal-links

You can verify if it's all setup by using this tool: https://limitless-sierra-4673.herokuapp.com/. Ignore the IPA file upload and just paste your link in there. 

Please let me know if this doesn't work!

**Comment by mmilosevic** [comment]
Hi, @aaustin,

I'm experiencing same problem on my devices with iOS 9.2 and iOS 9.0.1. I went through [this](https://dev.branch.io/recipes/branch_universal_links/ios/#prerequisites-for-using-universal-links) document for Universal link setup and do everything regarding: 
- APP ID's, 
- Provisioning Profiles, 
- xCode, 
- entitlements file target
- Branch.io Dashboard

I am using Universal link in format `https://bnc.lt/«four-letter-identifier»/«link-hash»` and both devices 100% of the time navigate me to app store. We did not set any custom domains in dashboard, we are using only `https://bnc.lt/...` (in Capabilities tab of target settings I set `applinks:bnc.lt`)

Delegate method 
`- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void(^)(NSArray * __nullable restorableObjects))restorationHandler NS_AVAILABLE_IOS(8_0);` 
in `AppDelegate.m` is never called. Do you have any suggestion where to look, or what to check?

**Comment by subinrevi** [comment]
Hi, @aaustin,
                      I am using the domain provided by branch only as we don't have any sub domain. The weird thing is that it was working perfectly well in iOS9 as well and only when I changed the branch key to live I began facing these issues.

**Comment by aaustin** [comment]
When you guys test this, can you confirm that you're not pasting in the link to Safari or Chrome? Apple doesn't support pasted-in Universal Links unfortunately. You have to actually click it. I'd suggest from messages, notes or slack or something. Here's a list of the supported browsers:
https://dev.branch.io/recipes/branch_universal_links/ios/#which-appsbrowsers-support-universal-links

In the meantime, I'll review both of your app settings and will get back to you with findings.

**Comment by mmilosevic** [comment]
I was testing by clicking on link in slack.

**Comment by aaustin** [comment]
Can you try pasting it into Messages or Notes, then long pressing the link and see if the option to 'Open in Porter and Sail' appears? I wonder if perhaps you accidentally disabled Ulinks on that device. (Yes.. that's a thing... thanks Apple)

**Comment by subinrevi** [comment]
Hi @aaustin, 
                       I was testing it by clicking the link from imesssage and mail. It would be great if you can check the app settings. The app Id associated with my project is getonthewaveapp@gmail.com.

**Comment by aaustin** [comment]
@subinrevi Did you try the long press thing I mentioned in the previous comments?

If that didn't work, can you confirm these settings in your developer.apple.com account?
<img width="248" alt="screen shot 2015-12-14 at 9 20 24 pm" src="https://cloud.githubusercontent.com/assets/1332632/11802698/8612e860-a2a8-11e5-8c61-12b869f9a375.png">

Lastly, can you paste in the link you're testing with? I can verify the link you're using is supported.

**Comment by mmilosevic** [comment]
@aaustin in screenshot you can check what options are presented to me

![img_2149](https://cloud.githubusercontent.com/assets/4363556/11806833/085031d0-a316-11e5-9ad1-af7ad5542c5c.jpg)

**Comment by subinrevi** [comment]
Hi @aaustin, 

      The bundle Id & Apple team Id are the same as that in the developer account. When i tried long pressing the link the following items are shown in popup (attaching the image of that as well). When i used the test_key it was working well in iOS9 as well, but i began facing issues only when i changed the key to live. I have also created a new bundle Id but I have made the corresponding changes in the dashboard, app plist and entitlements file.
![img_2066](https://cloud.githubusercontent.com/assets/8567369/11809443/ecc54be4-a34c-11e5-8f33-727c06e7da11.PNG)

**Comment by subinrevi** [comment]
Following are the links i am using for testing 

1. https://bnc.lt/z4Wg/Q1NwCl6gip

2. https://bnc.lt/z4Wg/Q1NwCl6gip

**Comment by mmilosevic** [comment]
@aaustin it is working when I sign app with distribution certificate (distribute over TestFlight), but when I install app from xCode or distribute over Crashlytics it's not.

**Comment by mmilosevic** [comment]
@aaustin I found what was a problem (provisioning profiles).

Thanks for your effort.

**Comment by aaustin** [comment]
@subinrevi Please select 'Open in Wave'. The fact that this open is present in the list means that the links are configured correctly.

Once you click it, this will re-enable Universal Links. Beware though! It's far to easy to disable them: you only need to press the domain at the top right of the screen after the link opens the app ('bnc.lt' button). Once you've re-enabled it, all should work fine.

**Comment by subinrevi** [comment]
Hi milosevic,

                      Is it working for you in iOS9 now ?

**Comment by mmilosevic** [comment]
@subinrevi yes it is working now. Problem was different bundle identifier. The one set in branch.io dashboard was different from one that was installed with provisioning profile on device.

**Comment by subinrevi** [comment]
Hi @aaustin,

                   Disabling and enabling the link is device specific right? When i enable that link by long press its working, but if i don't do that it will take the user to the URL provided. Is there any option to set it enabled by default or does every user has to do this initially ? Does changing the bundle Id in Dashboard has any effect (i mean if I change it on Dashboard and plist and then change back there is some weird behaviour, not sure still checking).

**Comment by aaustin** [comment]
Hey Subin - It's the default behavior on all phones. However, Apple made it way too easy to disable. You just have to click on 'bnc.lt' at the top right after being Universally Linked. This will disable them on the phone until you do the long press. Unfortunately, this is an Apple thing and we have no control over it.

**Comment by aaustin** [comment]
I'm going to close this issue since I believe it's resolved!

**Comment by rsahu34** [comment]
Hi,
I am facing a issue with branch universal deep linking, when i clicking a link its redirect to my app through Safari browser. I flowed the all step on provided document " https://dev.branch.io/getting-started/universal-app-links/guide/ " .
And also followed this link "https://dev.branch.io/getting-started/universal-app-links/support/#troubleshooting-universal-links".

I've added the entitlements, updated the provisioning profile, and everything's setup.

When I run the app on my device, and open the link, this always redirect through Safari.

Any idea why this might be happening .. ?
